src/main/java/hudson/plugins/gradle/injection/GradleBuildScanInjection.java

```diff
@@ -35,111 +53,149 @@ public class GradleBuildScanInjection implements BuildScanInjection {
+        if (ccudPluginVersion != null && InjectionUtil.isOk(InjectionConfig.checkVersion(ccudPluginVersion))) {
+            EnvUtil.setEnvVar(node, JENKINSGRADLEPLUGIN_CCUD_PLUGIN_VERSION, ccudPluginVersion);
+        }
+    }
 
-        return Objects.equals(existingFileDigest, initScriptDigest.get());
+    private boolean initScriptChanged(FilePath gradleInitScriptFile) throws IOException, InterruptedException {
+        if (!gradleInitScriptFile.exists()) {
+            return true;
+        }
+        String existingFileDigest = gradleInitScriptFile.digest();
+        return !Objects.equals(existingFileDigest, initScriptDigest.get());
     }
 
-    private void removeInitScript(VirtualChannel channel, String initScriptDirectory) {
+    private void cleanup(Node node, String initScriptDirectory) {
         try {
-            FilePath gradleInitScriptFile = getInitScriptFile(channel, initScriptDirectory);
-            if (gradleInitScriptFile.exists()) {
-                LOGGER.fine("delete init script file");
-                if (!gradleInitScriptFile.delete()) {
-                    throw new IllegalStateException("Error while deleting init script");
-                }
-            }
+            removeInitScript(node.getChannel(), initScriptDirectory);
+            removeEnvironmentVariables(node);
         } catch (IOException | InterruptedException e) {
             throw new IllegalStateException(e);
         }
     }
 
-    private static FilePath getInitScriptFile(VirtualChannel channel, String initScriptDirectory) {
-        if (initScriptDirectory == null) {
-            throw new IllegalStateException("init script directory is null");
+    private void removeInitScript(VirtualChannel channel, String initScriptDirectory) throws IOException, InterruptedException {
+        FilePath gradleInitScriptFile = getInitScriptFile(channel, initScriptDirectory);
+        if (gradleInitScriptFile.exists()) {
+            LOGGER.info("Deleting Gradle init script " + gradleInitScriptFile.getRemote());
+
+            boolean deleted = gradleInitScriptFile.delete();
+            Preconditions.checkState(deleted, "Error while deleting init script");
         }
-        return new FilePath(channel, initScriptDirectory + "/" + GRADLE_INIT_FILE);
     }
 
-    private static boolean isGradleEnterpriseUrlSet(EnvVars envGlobal) {
-        return EnvUtil.isSet(envGlobal, JENKINSGRADLEPLUGIN_GRADLE_ENTERPRISE_URL);
+    private void removeEnvironmentVariables(Node node) {
+        ALL_INJECTED_ENVIRONMENT_VARIABLES.forEach(v -> EnvUtil.removeEnvVar(node, v));
+    }
+
+    private static FilePath getInitScriptFile(VirtualChannel channel, String initScriptDirectory) {
+        Preconditions.checkState(initScriptDirectory != null, "init script directory is null");
+
+        return new FilePath(channel, filePath(initScriptDirectory, GRADLE_INIT_FILE));
+    }
+
+    private static String filePath(String... parts) {
+        return String.join("/", parts);
```

Comment:
❓ is this fine in a windows environment ?

Reply:
Yes, because in the end we pass it to the internal Jenkins `FilePath` which will handle it accordingly to the platform.